### PR TITLE
refactor(api-service): bill platform usage from workflow_run_count MV fixes NV-8497

### PR DESCRIPTION
**Pull request by djabarovgeorge**
## Summary

- Add `WorkflowRunCountRepository.getPlatformUsageByDateRange` querying the `workflow_run_count` SummingMergeTree for `workflow_run_status_processing` events with UTC calendar-day bounds derived from half-open Date ranges.
- Delete `WorkflowRunRepository.getPlatformUsageByDateRange` (raw `workflow_runs FINAL` scan).
- Update billing e2e mock to inject the count repository.
- Enterprise submodule tip points at the companion billing usecase refactor.

```mermaid
flowchart LR
  CreateUsageRecords --> GetPlatformNotificationUsage
  GetSubscription --> GetPlatformNotificationUsage
  GetPlatformNotificationUsage -->|ClickHouse flag| WorkflowRunCountRepository
  WorkflowRunCountRepository --> MV["workflow_run_count MV"]
  GetPlatformNotificationUsage -->|default / shadow| Mongo["notifications Mongo"]
```

## Why

Platform billing usage was scanning `workflow_runs FINAL`, which is expensive at scale. The MV already aggregates processing events by day/org and is the right source for day-aligned Stripe usage records and subscription period counts.

## Companion PR

Enterprise: https://github.com/novuhq/packages-enterprise/pull/481  
Merge the enterprise PR first, then this one.

## Test plan

- [x] Unit: `workflow-run-count.repository.spec.ts` (query shape, org filter, empty result, midnight exclusive end)
- [ ] CI: billing e2e `get-platform-notification-usage.e2e-ee.ts`
- [ ] Re-enable shadow flag in prod and compare Mongo vs MV before flipping `IS_BILLING_USAGE_CLICKHOUSE_ENABLED`
- [ ] Validate Submodule Sync expected to fail until enterprise PR merges (do not "fix")

## Notes

- Existing flags unchanged: `IS_BILLING_USAGE_CLICKHOUSE_SHADOW_ENABLED` / `IS_BILLING_USAGE_CLICKHOUSE_ENABLED`
- Day bounds: callers pass half-open `[start, end)`; query uses `date >= toDate(start) AND date <= toDate(end - 1ms)` so midnight period ends do not overlap the next period

<h3>Greptile Summary</h3>

The PR moves platform billing usage aggregation from raw `workflow_runs FINAL` scans to the daily `workflow_run_count` materialized view.

- Adds organization-level aggregation of processing events over half-open UTC date ranges.
- Removes the previous raw workflow-run query.
- Updates repository tests and enterprise billing test wiring for the count repository.
- Advances the enterprise submodule to the companion billing refactor.

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

The previously reported adjacent-period overlap is fixed by converting the half-open end timestamp minus one millisecond to the inclusive UTC end day, and the new test verifies the midnight period-boundary case; no blocking failure remains.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libs/application-generic/src/services/analytic-logs/workflow-run-count/workflow-run-count.repository.ts | Adds materialized-view billing aggregation and correctly maps a midnight-exclusive period end to the preceding UTC day. |
| libs/application-generic/src/services/analytic-logs/workflow-run-count/workflow-run-count.repository.spec.ts | Covers query shape, organization filtering, empty results, and the previously reported midnight boundary overlap. |
| libs/application-generic/src/services/analytic-logs/workflow-run/workflow-run.repository.ts | Removes the superseded raw workflow-runs platform usage scan. |
| apps/api/src/app/billing/e2e/get-platform-notification-usage.e2e-ee.ts | Updates the enterprise billing use-case mock dependency to the count repository. |
| .source | Advances the enterprise submodule to the companion billing implementation. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  Billing[Billing usage use cases] --> Usage[GetPlatformNotificationUsage]
  Usage --> CountRepo[WorkflowRunCountRepository]
  CountRepo --> MV[(workflow_run_count MV)]
  CountRepo --> Bounds[Half-open range to inclusive UTC days]
```

<sub>Reviews (3): Last reviewed commit: ["chore: update subproject reference to la..."](https://github.com/novuhq/novu/commit/e49e4bd7479d2a1681a251b7f9aa64fa40680892) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49587575)</sub>

### CODE REVIEW

File: libs/application-generic/src/services/analytic-logs/workflow-run-count/workflow-run-count.repository.ts [hunk 1]
```diff
@@ -172,6 +172,53 @@ export class WorkflowRunCountRepository extends LogRepository<typeof workflowRun
...
+
+    const query = `
+      SELECT
+        organization_id,
+        sum(count) as count
+      FROM ${WORKFLOW_RUN_COUNT_TABLE_NAME}
+      WHERE
+        date >= {startDate:Date}
+        AND date <= {endDate:Date}
+        AND event_type = 'workflow_run_status_processing'
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply libs/application-generic/src/services/analytic-logs/workflow-run-count/workflow-run-count.repository.ts:201]: Fixed in e69ed76f3a: callers pass a half-open `[start, end)`, so the MV query now uses `toDate(end - 1ms)` as the inclusive end day. Midnight Stripe period ends no longer pull in the next period’s first day. Spec covers that case; enterprise shadow logs updated in bf008dc to match.
